core/src/avm1/globals/file_reference_list.rs

```diff
@@ -0,0 +1,141 @@
+
+const PROTO_DECLS: &[Declaration] = declare_properties! {
+    "browse" => method(browse; DONT_ENUM);
+};
+
+const OBJECT_DECLS: &[Declaration] = declare_properties! {};
+
+pub fn browse<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let file_filters = match args.get(0) {
+        Some(Value::Object(array)) => {
+            // Array of filter objects.
+            let length = array.length(activation)?;
+
+            // Empty array is not allowed
+            if length == 0 {
+                return Ok(false.into());
+            }
+
+            let mut results = Vec::with_capacity(length as usize);
+
+            for i in 0..length {
+                if let Value::Object(element) = array.get_element(activation, i) {
+                    let mac_type = if let Ok(val) = element.get("macType", activation) {
+                        Some(val.coerce_to_string(activation)?.to_string())
+                    } else {
+                        None
+                    };
+
+                    let description = element
+                        .get("description", activation)?
+                        .coerce_to_string(activation)?
+                        .to_string();
+
+                    let extensions = element
+                        .get("extension", activation)?
+                        .coerce_to_string(activation)?
+                        .to_string();
+
+                    // Empty strings are not allowed for desc / extension
+                    if description.is_empty() || extensions.is_empty() {
+                        return Ok(false.into());
+                    }
+
+                    results.push(FileFilter {
+                        description,
+                        extensions,
+                        mac_type,
+                    });
+                } else {
+                    return Err(Error::ThrownValue("Unexpected filter value".into()));
+                }
+            }
+
+            results
+        }
+        None => Vec::new(),
+        _ => return Ok(Value::Undefined),
+    };
+
+    let dialog = activation
+        .context
+        .ui
+        .display_file_open_dialog(file_filters, true);
+
+    let result = match dialog {
+        Some(dialog) => {
+            let process = activation.context.load_manager.select_multiple_file_dialog(
+                activation.context.player.clone(),
+                this,
+                dialog,
+            );
+
+            activation.context.navigator.spawn_future(process);
+            true
+        }
+        None => false,
+    };
+
+    Ok(result.into())
+}
+
+pub(crate) fn constructor<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    this.set_native(
+        activation.context.gc_context,
+        NativeObject::FileReference(FileReferenceObject(GcCell::new(
```

Comment:
Shouldn't this be `FileReferenceList` and `FileReferenceListObject`? Is this perhaps a copy-paste bug? :thinking: